packages/pwa/app/hooks/use-add-to-cart-modal.js

```diff
@@ -22,33 +22,40 @@ import {
     Stack,
     useBreakpointValue
 } from '@chakra-ui/react'
-import Link from '../link'
-import useBasket from '../../commerce-api/hooks/useBasket'
-import {LockIcon} from '../icons'
-import {useCurrency} from '../../hooks'
-import {useVariationAttributes} from '../../hooks'
-import {findImageGroupBy} from '../../utils/image-groups-utils'
+import useBasket from '../commerce-api/hooks/useBasket'
+import Link from '../components/link'
+import RecommendedProducts from '../components/recommended-products'
+import {LockIcon} from '../components/icons'
+import {DEFAULT_CURRENCY} from '../constants'
+import {useVariationAttributes} from './'
+import {findImageGroupBy} from '../utils/image-groups-utils'
+
+export const AddToCartModalContext = React.createContext()
+export const useAddToCartModalContext = () => useContext(AddToCartModalContext)
```

Comment:
Should we declare this context inside the `index.js` under the `context` folder? How should we decide which context will stay in its own file or inside the context file?

Reply:
I noticed `develop` branch has a bunch of changes made to `context.js` (the update category context and the new currency context). Seems like we already established a pattern there, i guess i'll follow the  pattern and maybe we can chat a bit more in tech discussion session.

TL;DR i'm moving this to context.js

Reply:
I couldn't do this because there is a circular dependency between the context, hook and the component... I really don't want to separate the component from the hook and contexts because the component itself doesn't work, it is meant to work only with the hook.

Reply:
Would you mind adding a comment to explain that so it won't confuse other developers?